sdk/core/azure-core/src/main/java/com/azure/core/implementation/AccessTokenCache.java

```diff
@@ -137,7 +137,7 @@ private Supplier<Mono<? extends AccessToken>> retrieveToken(TokenRequestContext
                         return Mono.just(cache);
                     } else {
                         // wait for refreshing thread to finish but defer to updated cache in case just missed onNext()
-                        return sinksOne.asMono().switchIfEmpty(Mono.defer(() -> Mono.just(cache)));
+                        return sinksOne.asMono().switchIfEmpty(Mono.just(cache));
```

Comment:
Mono.just() is evaluated at assembly time while Mono.defer is evaluated at subscription time. So, should we continue to keep this as Mono.defer() to wait for the cache to be updated at subscription?

Reply:
It won't be same as `Mono.just`.  `cache` is volatile field in the `AccessTokenCache`

Reply:
> It won't be same as `Mono.just`. `cache` is volatile field in the `AccessTokenCache`

Ah, okay, didn't notice that, I'll change it to `Mono.fromSupplier`.